### PR TITLE
Fix issue HasSnapshot() cannot find last snapshot

### PR DESCRIPTION
**Pull request by shuo-wu**
https://github.com/rancher/longhorn/issues/568

### CODE REVIEW

File: replica/backup.go [hunk 1]
```diff
@@ -77,7 +73,7 @@ func (rb *Backup) OpenSnapshot(snapID, volumeID string) error {
 
 func (rb *Backup) assertOpen(id, volumeID string) error {
```
Comment (shuo-wu) [inline replica/backup.go]: Why do we need `assertOpen()`? This function restricts that we can only operate one snapshot for each `Backup` object. 
e.g. If I use the backup object (it's `deltaOps` in backupstore) to open snapshot A (`OpenSnapshot()`), then later I could read snapshot A only (`ReadSnapshot()`), or check the existence of snapshot A only (`HasSnapshot()`). It means we cannot use this backup object to check/read snapshot B.

**Comment by yasker** [comment]
Patches are LGTM. Need to find a way to test this, make sure it won't happen again.

**Comment by yasker** [comment]
@shuo-wu We can test it by adding a parameter e.g. `IncrementalOnly` into DeltaBackupConfig, which makes the function errors out if that's not the case. This way we can know if a full backup or incremental backup in our tests.

**Comment by yasker** [comment]
Merged this for now. The testing is tracking at #592.

**Pull request merged**